### ISSUE TITLE
[0.11.0] Malformed Select statement causes panic

### ISSUE THREAD

**Issue by mjdesa** [influxdata/influxdb#5962]
To repro run `influx_stress` then issue the following commands in the `influx` CLI.

The query
```
> select count("*") * "*" from cpu
```

induces the following panic:
```
[query] 2016/03/09 22:36:10 SELECT count("*") * "*" FROM stress."default".cpu
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x6eb986]

goroutine 102 [running]:
github.com/influxdata/influxdb/influxql.Iterators.Close(0xc20ad564e0, 0x1, 0x1, 0x0, 0x0)
        /tmp/influxdb/src/github.com/influxdata/influxdb/influxql/iterator.go:39 +0x96
github.com/influxdata/influxdb/influxql.(*Emitter).Close(0xc20b566280, 0x0, 0x0)
        /tmp/influxdb/src/github.com/influxdata/influxdb/influxql/emitter.go:38 +0x4d
github.com/influxdata/influxdb/cluster.(*QueryExecutor).executeSelectStatement(0xc20805a6c0, 0xc20e38edd0, 0x2710, 0x0, 0xc20b4720c0, 0xc20b472060, 0x0, 0x0)
        /tmp/influxdb/src/github.com/influxdata/influxdb/cluster/query_executor.go:510 +0xe4b
github.com/influxdata/influxdb/cluster.(*QueryExecutor).executeQuery(0xc20805a6c0, 0xc20b4ea3c0, 0xc2080fdf8e, 0x6, 0x2710, 0xc20b472060, 0xc20b4720c0)
        /tmp/influxdb/src/github.com/influxdata/influxdb/cluster/query_executor.go:120 +0x6ce
created by github.com/influxdata/influxdb/cluster.(*QueryExecutor).ExecuteQuery
        /tmp/influxdb/src/github.com/influxdata/influxdb/cluster/query_executor.go:73 +0x8c

goroutine 1 [chan receive]:
main.(*Main).Run(0xc208039700, 0xc20800a010, 0x4, 0x4, 0x0, 0x0)
        /tmp/influxdb/src/github.com/influxdata/influxdb/cmd/influxd/main.go:94 +0x765
main.main()
        /tmp/influxdb/src/github.com/influxdata/influxdb/cmd/influxd/main.go:45 +0xdc
```

[gist with full panic](https://gist.github.com/mjdesa/77f979058035a595b64e)

**Comment by mjdesa** [comment]
Actually it seems like trying to do any math on a non-existing field yields a panic.

```
> select value + hello from cpu
ERR: Get http://localhost:8086/query?db=stress&epoch=ns&q=select+value+%2B+hello+from+cpu: EOF
```

**Comment by e-dard** [comment]
@mjdesa In the case of arbitrary math on fields however, I think the panic is different. I'm seeing this for arbitrary math:

```
[query] 2016/03/09 23:22:40 SELECT value + lol FROM db."default".cpu
[0xc820992420]
panic: send on closed channel

goroutine 106 [running]:
panic(0x6c8640, 0xc8209de3f0)
	/usr/local/share/go/src/runtime/panic.go:464 +0x3e6
github.com/influxdata/influxdb/influxql.auxIteratorFields.send(0xc8207aa740, 0x2, 0x2, 0x1412e30, 0xc82055e420)
	/Users/edd/Documents/projects/golang/src/github.com/influxdata/influxdb/influxql/iterator.go:354 +0xdab
github.com/influxdata/influxdb/influxql.(*floatAuxIterator).stream(0xc820634870)
	/Users/edd/Documents/projects/golang/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:625 +0x107
created by github.com/influxdata/influxdb/influxql.(*floatAuxIterator).Start
	/Users/edd/Documents/projects/golang/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:588 +0x35
```

**Comment by mjdesa** [comment]
@e-dard ah you're right. I didn't check the panic and just assumed it was the same one.

**Comment by mjdesa** [comment]
I'll change the name of the issue again.
@e-dard should I open another issue for arbitrary math on fields as well?

**Comment by e-dard** [comment]
Yeah as the fix might be different.

**Comment by jsternberg** [comment]
#5979 will fix this partially and #5990 should make sure this binary expression isn't valid to begin with.

I also keep seeing things like `SELECT count("*") FROM cpu` being used and it doesn't seem to be valid syntax. We don't yet support wildcards inside of function calls and using a wildcard would be `count(*)`. I made a quick check to see how the parser handled `count("*")` and it treats the argument as a field instead of a wildcard. The problem here might have been it trying to use a field that didn't exist which wasn't working.

**Comment by jsternberg** [comment]
Just merged into master and 0.11.

```
> select count("*") * "*" from cpu
ERR: error parsing query: binary expressions cannot mix aggregates and raw fields
```
